config.go

```diff
@@ -28,47 +27,43 @@ type config struct {
 	button  *configButton
 }
 
-func prepareConfig() *config {
-
-	mqttServer, ok := os.LookupEnv("MQTT_SERVER")
-
-	if !ok {
-		mqttServer = "tcp://192.168.1.194:1883"
-	}
-
-	mqttTopic, ok := os.LookupEnv("MQTT_TOPIC")
+// Environment interface mimics the existing os package and the dependencies
+// we need for loading the config
+type Environment interface {
+	LookupEnv(key string) (string, bool)
+}
 
-	if !ok {
-		mqttTopic = "godor/door1"
-	}
+func prepareConfig() *config {
 
-	lePin, ok := os.LookupEnv("GODOR_PIN")
-	if !ok {
-		lePin = "1"
-	}
+	gpioPin := env("GODOR_PIN", "1")
 
-	leIntPin, err := strconv.Atoi(lePin)
+	leIntPin, err := strconv.Atoi(gpioPin)
 	if err != nil {
-		panic("Wrong pin")
+		panic("unable to convert pin number")
 	}
 
-	log.Println(mqttTopic)
-	log.Println(lePin)
-
 	return &config{
 		emulate: true,
 		mqtt: &configMqtt{
-			server: mqttServer,
-			topic:  mqttTopic,
+			server: env("MQTT_SERVER", "tcp://192.168.1.194:1883"),
+			topic:  env("MQTT_TOPIC", "godor/door1"),
 		},
 
 		sensor: &configSensor{
 			pin:       leIntPin,
 			checkRate: time.Second,
 		},
 
+		// not used yet
 		button: &configButton{
 			pin: 1,
 		},
 	}
 }
+
+func env(key, defaultValue string) string {
+	if val, ok := os.LookupEnv(key); ok {
```

Comment:
I see whatthis line is doing, unsure how it works though. Whats the `ok` at the end of the LookupEnv call?

Reply:
ok so lets break it down
```golang
val, ok := os.LookupEnv(key);
```
this is the same as before. LookupEnv returns two values, the string value of the env + an boolean that represents if the ENV actually exists.

with golang you can have mulltiple statements within an `if()`. The last operation is what is evaluated. So if we were to re-write this in a more verbose way it would be

```golang
val, ok := os.LookupEnv(key)
if (ok) {
    return val;
}
return defaultValue
```